src/contradictory_claims/data/preprocess_cord.py

```diff
@@ -199,3 +204,51 @@ def clean_text(input_data):
     input_processed = input_data.loc[sentences_to_keep, :]
 
     return input_processed
+
+
+def merge_section_text(input_data: pd.DataFrame):
+    """
+    Merge all sentences belonging to each section of each paper into contiguous text passages.
+
+    :param input_data: pandas dataframe with publication text
+    :return: Dataframe with merged section text
+    """
+    # Merge all sentences belonging to each section of each paper into contiguous text passages
+    merged_df = input_data.groupby(['cord_uid', 'section'], as_index=False).agg({'sentence': ' '.join})
+    # Convert text to lower case
+    merged_df.loc[:, 'sentence'] = merged_df.loc[:, 'sentence'].str.lower()
+    # Rename column
+    merged_df = merged_df.rename(columns={'sentence': 'text'})
+
+    return merged_df
+
+
+def filter_section_with_drugs(input_data: pd.DataFrame, drug_terms: List[str], drug_terms_pattern: str):
+    """
+    Filter to sections where section text contains drug terms.
+
+    :param input_data: pandas dataframe with publication text
+    :param drug_terms: list of drug terms to search for
+    :param drug_terms_pattern: drug terms regex pattern to search for
+    :return: Dataframe with sections containing drug terms
+    """
+    # Replace drug name short forms with full forms
+    drug_replace_dict = {'hcq': 'hydroxychloroquine', ' cq ': 'chloroquine', ' azt ': 'azithromycin',
+                         ' azi ': 'azithromycin', ' az ': 'azithromycin'}
+    for key, value in drug_replace_dict.items():
+        input_data['text'] = [t.lower().replace(key, value) for t in input_data.text]
+
+    # Filter to sections where section text contains drug terms
+    contain_drug_mask = input_data['text'].str.contains(drug_terms_pattern, case=False)
+    drugs_section_df = input_data[contain_drug_mask]
+
+    # Add a new column for storing the drug term matches
+    drugs_section_df['drug_terms_used'] = ''
+
+    # Flank drug terms with white space for accurate match
+    drug_terms = [' ' + d + ' ' for d in drug_terms]
```

Comment:
Yeah I know, it's a difficult call to make...Michael did some analysis on this and noted that not checking for spaces brings in a lot of noise. I did check a few cases myself and I think it's wiser to check for spaces. Ex of noise: 
1. stimate -> estimate
2. collagen -> collagenous
3. ultane -> simultaneouly
4. differin -> differing
5. propel -> propelling
6. compro -> compromised

Alternatively, we can refine the drug terms list because many of the noise producing drug names seem like stuff we wouldn't really care about anyway..but that needs more informed eyes than mine..